common/test/views/support/cleaner/AmpEmbedCleanerTest.scala

```diff
@@ -140,76 +156,203 @@ class AmpEmbedCleanerTest extends FlatSpec with Matchers {
     val result: Document = clean(document)
-
     result.getElementsByTag("amp-soundcloud").size should be(0)
   }
 
+
+
+  /////////////////////////////
+  // Audio Embed cleaner
+  /////////////////////////////
+
+  "AmpEmbedCleaner" should "replace an iframe in an audio-element that is not a soundcloud embed with an amp-iframe element" in {
+    val doc = <html>
+                  <body>
+                    <figure class="element-audio">
+                      <iframe frameborder="0" width="460" height="300" src={audioBoomUrl}></iframe>
+                    </figure>
+                  </body>
+              </html>.toString()
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+    result.getElementsByTag("amp-iframe").size should be(1)
+  }
+
+  "AmpEmbedCleaner" should "not add an amp-iframe element if an audio embed contains an iframe with src url from soundcloud" in {
+    val doc = <html>
+                  <body>
+                    <figure class="element-audio">
+                      <iframe src={soundcloudUrlV1}></iframe>
+                    </figure>
+                  </body>
+              </html>.toString()
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+    result.getElementsByTag("amp-iframe").size should be(0)
+  }
+
+  "AmpEmbedCleaner" should "not add an amp-iframe element if an audio element does not contain an iframe" in {
+    val doc = <html>
+                  <body>
+                    <figure class="element-audio"></figure>
+                  </body>
+              </html>.toString()
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+    result.getElementsByTag("amp-iframe").size should be(0)
+  }
+
+  "AmpEmbedCleaner" should "create an amp-iframe element with a data-main-player-id from the iframe src from an audioboom embed" in {
+    val doc = <html>
+                  <body>
+                    <figure class="element-audio">
+                      <iframe frameborder="0" width="460" height="300" src={audioBoomUrl}></iframe>
+                    </figure>
+                  </body>
+              </html>.toString()
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+    result.getElementsByTag("amp-iframe").first.attr("src") should be(audioBoomUrl)
+  }
+
+
+
   /////////////////////////////
   // Maps cleaner
   /////////////////////////////
 
   "AmpEmbedCleaner" should "replace an iframe in an map element with an amp-iframe element" in {
-    val doc = s"""<html><body><figure class="element-map"><iframe src="$googleMapsUrl"></iframe></figure></body></html>"""
+    val doc = <html>
+                  <body>
+                    <figure class="element-map">
+                      <iframe src={googleMapsUrl}></iframe>
+                    </figure>
+                  </body>
+              </html>.toString()
```

Comment:
Can we extract this into a method? A similar pattern is used in a number of tests